File: FishBun/src/main/java/com/sangcomz/fishbun/BaseProperty.java
package com.sangcomz.fishbun;

import android.net.Uri;

import java.util.ArrayList;

/**
 * Created by sangcomz on 13/05/2017.
 */

interface BaseProperty {
    FishBunCreator setArrayPaths(ArrayList<Uri> arrayPaths);

    FishBunCreator setPickerCount(int count);

    FishBunCreator setMaxCount(int count);

    FishBunCreator setMinCount(int count);

    FishBunCreator setRequestCode(int RequestCode);

    FishBunCreator setReachLimitAutomaticClose(boolean isAutomaticClose);

    FishBunCreator exceptGif(boolean isExcept);

    void startAlbum();
}

File: FishBun/src/main/java/com/sangcomz/fishbun/BaseProperty.kt
package com.sangcomz.fishbun

import android.content.Intent
import android.net.Uri
import androidx.activity.result.ActivityResultLauncher

/**
 * Created by sangcomz on 13/05/2017.
 */
interface BaseProperty {
    fun setSelectedImages(selectedImages: ArrayList<Uri>): FishBunCreator

    fun setPickerCount(count: Int): FishBunCreator

    fun setMaxCount(count: Int): FishBunCreator

    fun setMinCount(count: Int): FishBunCreator

    fun setRequestCode(requestCode: Int): FishBunCreator

    fun setReachLimitAutomaticClose(isAutomaticClose: Boolean): FishBunCreator

    fun exceptGif(isExcept: Boolean): FishBunCreator

    fun exceptMimeType(exceptMimeTypeList: List<MimeType>): FishBunCreator

    fun startAlbum()

    fun startAlbumWithOnActivityResult(requestCode: Int)

    fun startAlbumWithActivityResultCallback(activityResultLauncher: ActivityResultLauncher<Intent>)
}